src/WebJobs.Extensions.DocumentDB/Bindings/DocumentDBAttributeBindingProvider.cs

```diff
@@ -0,0 +1,118 @@
+                    DocumentDBConfiguration.AzureWebJobsDocumentDBConnectionStringName));
+            }
+
+            DocumentDBContext documentDBContext = CreateContext(_docDBConfig, attribute, _jobHostConfig.NameResolver);
+
+            if (attribute.CreateIfNotExists)
+            {
+                await CreateIfNotExistAsync(documentDBContext.Service, documentDBContext.ResolvedDatabaseName, documentDBContext.ResolvedCollectionName);
+            }
+
+            Type coreType = DocumentDBUtility.GetCoreType(parameter.ParameterType);
+
+            IBinding binding = null;
+            try
+            {
+                binding = BindingFactory.BindGenericCollector(parameter, typeof(DocumentDBAsyncCollector<>), coreType,
+                _jobHostConfig.GetService<IConverterManager>(), (s) => documentDBContext);
+            }
+            catch (Exception ex)
+            {
+                // BindingFactory uses a couple levels of reflection so we get TargetInvocationExceptions here.
+                // This pulls out the root exception and rethrows it.
+                while (ex.InnerException != null)
+                {
+                    ex = ex.InnerException;
+                }
+
+                ExceptionDispatchInfo.Capture(ex).Throw();
+            }
+
+            return binding;
+        }
+
+        internal static DocumentDBContext CreateContext(DocumentDBConfiguration config, DocumentDBAttribute attribute, INameResolver resolver)
+        {
+            return new DocumentDBContext
+            {
+                Service = config.DocumentDBServiceFactory.CreateService(config.ConnectionString),
+                ResolvedDatabaseName = Resolve(attribute.DatabaseName, resolver),
+                ResolvedCollectionName = Resolve(attribute.CollectionName, resolver)
+            };
+        }
+
+        public static async Task CreateIfNotExistAsync(IDocumentDBService service, string databaseName, string documentCollectionName)
+        {
+            Uri databaseUri = UriFactory.CreateDatabaseUri(databaseName);
+            Database database = new Database { Id = databaseName };
+            DocumentCollection documentCollection = new DocumentCollection
+            {
+                Id = documentCollectionName
+            };
+
+            // If we queried for the Database or Collection before creation, we may hit a race condition
+            // if multiple instances are running the same code. So let's just create and ignore a Conflict.
+            await DocumentDBUtility.ExecuteAndIgnoreStatusCodeAsync(HttpStatusCode.Conflict,
+                () => service.CreateDatabaseAsync(database));
+
+            await DocumentDBUtility.ExecuteAndIgnoreStatusCodeAsync(HttpStatusCode.Conflict,
+                () => service.CreateDocumentCollectionAsync(databaseUri, documentCollection));
+        }
+
+        private static string Resolve(string value, INameResolver resolver)
```

Comment:
I see every single extension has to duplicate this. Probably something I should expose as a public helper/extension method. Not for now, just bringing it up. Maybe log a bug for it.